src/app/in-memory-data.service.ts

```diff
@@ -327,6 +327,89 @@ export class InMemoryDataService implements InMemoryDbService {
       'imageURL': 'https://avatars.githubusercontent.com/u/2410474?v=3'
     };
 
-    return {loginStatus, workitems, workitemtypes, user};
+    let linkCategories = {
+      "meta": { "comment": "Note, that the 'id' field is optional for a create request." },
+      "id": "1",
+      "type": "linkcategories",
+      "attributes": {
+        "name": "test",
+        "title": "This category is reserved for link types that belong to the 'test' subsystem."
+      }
+    }
+
+    let linkTypes = [
```

Comment:
This is how the result of `GET /api/workitemlinktypes` looks like. See also http://swagger.goa.design/?url=kwk%2Falmighty-core%2Fdesign#!/work-item-link-type/work_item_link_type_list . **Notice, that `source_type` and `target_types` are now part of the relationships object. We don't have `links` objects with `self` and `related` members for every resource yet. We also store a `version` attribute that is used for concurrent updates. It is part of workitem links/types/categories.**

```json
{
  "data": [
    {
      "attributes": {
        "description": "A test work item can 'test' if a the code in a pull request passes the tests.",
        "forward_name": "test-workitemtype",
        "name": "tested-by-link-type",
        "reverse_name": "tested by",
        "version": 0
      },
      "id": "40bbdd3d-8b5d-4fd6-ac90-7236b669af04",
      "relationships": {
        "link_category": {
          "data": {
            "id": "6c5610be-30b2-4880-9fec-81e4f8e4fd76",
            "type": "workitemlinkcategories"
          }
        },
        "source_type": {
          "data": {
            "id": "system.bug",
            "type": "workitemtypes"
          }
        },
        "target_type": {
          "data": {
            "id": "system.bug",
            "type": "workitemtypes"
          }
        }
      },
      "type": "workitemlinktypes"
    }
  ],
  "meta": {
    "totalCount": 1
  }
}
```